### PR TITLE
[8] [9] [UI] As a user, I can see Products shown by grouped

### LINKED ISSUE #8: [UI] As a user, I can see Products shown by grouped
## Why

- On the Home screen, after a scroll view of product design, there are sections to list down products
 
## Acceptance Criteria

- A collection of Products is shown.
- Each section:
  - Title
  - Should contain maximum of 2 items
  - `Shop all` button on top left of section

## Resources

Screen Shot 2021-04-22 at 16 18 45

- [Design](https://www.figma.com/proto/GXmbnWUEH9ZkrUluaPyedn/e-commerce?node-id=48%3A68&amp%3Bviewport=1944%2C-468%2C1.197691559791565&amp%3Bscaling=scale-down&scaling=min-zoom)

### LINKED ISSUE #9: [UI] As a user, I can see a Product at Home screen
## Why

- On the Home screen, in the collection, a Product can show itself
 
## Acceptance Criteria

- Product image
- Title
- Price

## Resources

Screen Shot 2021-04-22 at 16 18 45

- [Design](https://www.figma.com/proto/GXmbnWUEH9ZkrUluaPyedn/e-commerce?node-id=48%3A68&amp%3Bviewport=1944%2C-468%2C1.197691559791565&amp%3Bscaling=scale-down&scaling=min-zoom)

### PR DESCRIPTION
**Pull request by vnntsu**
#8
#9

## What happened 👀

- A collection of Products is shown.
- Each section:
  - Title
  - Shop all button on top left of section

### CODE REVIEW

File: ecommerce-ios/Sources/Home/View/ProductCollectionView/ProductCollectionView.swift [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    private let viewModel = ProductCollectionViewViewModel()
+
+    private let numberOfColumns: Int = 2
+    private let spacing: CGFloat = 17.0
+
+    private var columns: [GridItem] {
+        let numberOfColumns = CGFloat(self.numberOfColumns)
+        let minimunWidth = ((screenWidth - spacing * (numberOfColumns + 1)) / numberOfColumns).rounded(.down)
+        return [.init(.adaptive(minimum: minimunWidth))]
+    }
```
Comment (blyscuit) [inline ecommerce-ios/Sources/Home/View/ProductCollectionView/ProductCollectionView.swift]: Please take a look at https://github.com/nimblehq/nimble-ecommerce-ios/pull/53#discussion_r635741188 and tell us what you think?

File: ecommerce-ios/Sources/Home/View/ProductCollectionView/ProductCollectionViewViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+//  ecommerce-ios
+//
+//  Created by Su T. Nguyen on 20/05/2021.
+//
+
+import Foundation
+
+struct ProductCollectionViewViewModel {
+
+    let sections: [ProductSection]
```
Comment (nmint8m) [inline ecommerce-ios/Sources/Home/View/ProductCollectionView/ProductCollectionViewViewModel.swift]: Please rename to sectionViewModels

File: ecommerce-ios/Sources/Home/View/ProductCollectionView/ProductSection.swift [hunk 3]
```diff
@@ -0,0 +1,46 @@
+//
+//  ProductSection.swift
+//  ecommerce-ios
+//
+//  Created by Su T. Nguyen on 20/05/2021.
+//
+
+import Foundation
+
+struct ProductSection: Identifiable {
```
Comment (nmint8m) [inline ecommerce-ios/Sources/Home/View/ProductCollectionView/ProductSection.swift]: Please rename to SectionViewModel

File: ecommerce-ios/Sources/Home/View/ProductCollectionView/ProductCollectionView.swift [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+                    HStack {
+                        Text(section.title)
+                            .font(.system(size: 22.0, weight: .bold))
+                        Spacer()
+                        Button("Shop all") {
+                            print("Did tap shop all button")
+                        }
+                        .font(.system(size: 15.0))
+                        .foregroundColor(.purpleBlue)
+                    }
```
Comment (nmint8m) [inline ecommerce-ios/Sources/Home/View/ProductCollectionView/ProductCollectionView.swift]: Please define a separated View for this. You can name it with ProductSectionHeaderView. I think we should make this one customizable with a title label and a trailing view. I suggest you to look at the Document how to customize view inside a view that I shared in Notion.

File: ecommerce-ios/Views/ProductCollectionView/ProductSectionViewModel.swift [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+        for index in 0..<collection.numberOfItems {
+             let cellVM = ProductCellViewModel(
+                id: index,
+                name: "Pink cube",
+                imageString: "screen-search-result-tshirt/tshirt-cube-pink",
+                price: 10_000,
+                currency: "฿"
+            )
+            cellVMs.append(cellVM)
+        }
```
Comment (nmint8m) [inline ecommerce-ios/Views/ProductCollectionView/ProductSectionViewModel.swift:32]: No need to use a `for` loop here 💯

```suggestion
        var cellViewModels = Array(
            repeating: ProductCellViewModel(
                id: index,
                name: "Pink cube",
                imageString: "screen-search-result-tshirt/tshirt-cube-pink",
                price: 10_000,
                currency: "฿"
            ), 
            count: collection.numberOfItems
        )
```

**Review by nmint8m — APPROVED** [review]

**Review by blyscuit — APPROVED** [review]
looks good

**Pull request merged**